modules/apps/foundation/roles/roles-admin-web/src/main/java/com/liferay/roles/admin/web/internal/portlet/RolesAdminPortlet.java

```diff
@@ -282,6 +283,11 @@ public void updateActions(
 
 		long roleId = ParamUtil.getLong(actionRequest, "roleId");
 
+		long groupId = themeDisplay.getScopeGroupId();
+
+		_permissionService.checkPermission(
+			groupId, Role.class.getName(), roleId);
```

Comment:
Seeing the next commit I'm guessing this might have been done for performance. Don't do that, it's better to have some performance penalty (which might even be negligible due to caching) than to spread permissions logic around.